packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -117,7 +117,7 @@ abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
 
   static final Map<GlobalKey, Element> _registry = <GlobalKey, Element>{};
   static final Set<Element> _debugIllFatedElements = HashSet<Element>();
-  static final Map<GlobalKey, Element> _debugReservations = <GlobalKey, Element>{};
+  static final Map<Element, Map<Element, GlobalKey>> _debugReservations = <Element, Map<Element, GlobalKey>>{};
```

Comment:
In stead of mapping
globalKey -> parent

I instead mapping
parent,child -> globalkey

In the case of early rebuild, I can correctly remove the reservation when i rebuild the node in layout stage.
During the layout rebuild, I will attempt to remove the early rebuild reservation in 154